src/parquet/util/bit-stream-utils.inline.h

```diff
@@ -75,7 +75,7 @@ inline bool BitWriter::PutAligned(T val, int num_bytes) {
   return true;
 }
 
-inline bool BitWriter::PutVlqInt(int32_t v) {
+inline bool BitWriter::PutVlqInt(uint32_t v) {
```

Comment:
I think the impala implementation is wrong and doesn't handle negative numbers. We want to do a logical shift, not arithmetic shift. The version with int32_t will spin forever for negative numbers.